sympy/solvers/solveset.py

```diff
@@ -539,23 +539,7 @@ def _solve_real_trig(f, symbol):
     """ Helper to solve trigonometric equations """
     f = trigsimp(f)
     f = f.rewrite(exp)
-    f = together(f)
-    g, h = fraction(f)
-    y = Dummy('y')
-    g, h = g.expand(), h.expand()
-    g, h = g.subs(exp(I*symbol), y), h.subs(exp(I*symbol), y)
-    if g.has(symbol) or h.has(symbol):
-        return ConditionSet(symbol, Eq(f, 0), S.Reals)
-
-    solns = solveset_complex(g, y) - solveset_complex(h, y)
-
-    if isinstance(solns, FiniteSet):
-        return Union(*[invert_complex(exp(I*symbol), s, symbol)[1]
-                       for s in solns])
-    elif solns is S.EmptySet:
-        return S.EmptySet
-    else:
-        return ConditionSet(symbol, Eq(f, 0), S.Reals)
```

Comment:
If modified  [this line ](https://github.com/sympy/sympy/pull/10482/files#diff-eec0422923e8f100745c015cd8fdd6cfR542) with `return solveset_real(f,symbol)`

Then it is giving solution in ` exp` form ,that is not we want , right?
```
>>> print solveset(sin(x + a) - sin(x), a, domain=S.Reals)
ConditionSet(a, Eq(-exp(2*I*x) - exp(I*x)*exp(I*(-a - x)) + exp(I*x)*exp(I*(a + x)) + 1, 0), (-oo, oo))
>>> print solveset(sin(x + a) - sin(x), a)
ConditionSet(a, Eq(-sin(x) + sin(a + x), 0), Complexes(Lambda((_x, _y), _x + _y*I), (-oo, oo) x (-oo, oo)))

```
we can `return solveset_complex(f,symbol))` for solution 
```
>>> solveset(sin(x + a) - sin(x), a)
{a | a ∊ ℂ ∧ -sin(x) + sin(a + x) = 0}
>>> solveset(sin(x + a) - sin(x), a, domain=S.Reals)
{2⋅n⋅π | n ∊ ℤ}
```
which is same as what `return solveset(f,symbol)` gives.